docs/agents/agent-development.md

```diff
@@ -237,14 +237,33 @@ Global labels can be specified via the environment variable `ELASTIC_APM_GLOBAL_
 
 ### Transactions
 
+#### Transaction outcome
+
+The `outcome` property denotes whether the transaction represents a success or a failure from the perspective of the entity that produced the event.
+The APM Server converts this to the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) field.
+This property is optional to preserve backwards compatibility.
+If an agent doesn't report the `outcome` (or reports `null`), the APM Server sets the outcome to `"unknown"`.
+
+- `"failure"`: Indicates that this transaction describes a failed result. \
+  Note that client errors don't fall into this category as they are not an error from the perspective of the server.
```

Comment:
This note seems confusing if the reader isn't aware that this is about HTTP client errors. You have a similar note in the "HTTP Transactions" section, so I'd remove this note (unless you mean other client/server environments, in which case I think the note should be expanded with examples)

Reply:
This is meant as a protocol-agnostic description. But an example definitely wouldn't hurt. What about just adding this:

```suggestion
  Note that client errors (such as HTTP 4xx) don't fall into this category as they are not an error from the perspective of the server.
```

Reply:
LGTM